vault/resource_pki_secret_backend_crl_config_test.go

```diff
@@ -7,125 +7,149 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 
 	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
 	"github.com/hashicorp/terraform-provider-vault/testutil"
 )
 
-func TestPkiSecretBackendCrlConfig_basic(t *testing.T) {
-	rootPath := "pki-root-" + strconv.Itoa(acctest.RandInt())
+func getCRLConfigChecks(resourceName string, isUpdate bool) resource.TestCheckFunc {
+	baseChecks := []resource.TestCheckFunc{
+		resource.TestCheckResourceAttr(resourceName, "expiry", "72h"),
+		resource.TestCheckResourceAttr(resourceName, "disable", "true"),
+	}
+	return func(state *terraform.State) error {
+		var checks []resource.TestCheckFunc
+		meta := testProvider.Meta().(*provider.ProviderMeta)
+		if meta.IsAPISupported(provider.VaultVersion112) {
+			var extras []resource.TestCheckFunc
+			if !isUpdate {
+				extras = []resource.TestCheckFunc{
+					resource.TestCheckResourceAttr(resourceName, "ocsp_disable", "false"),
+					resource.TestCheckResourceAttr(resourceName, "ocsp_expiry", "12h"),
+					resource.TestCheckResourceAttr(resourceName, "auto_rebuild", "true"),
+					resource.TestCheckResourceAttr(resourceName, "auto_rebuild_grace_period", "12h"),
+					resource.TestCheckResourceAttr(resourceName, "enable_delta", "true"),
+					resource.TestCheckResourceAttr(resourceName, "delta_rebuild_interval", "15m"),
+				}
+			} else {
+				extras = []resource.TestCheckFunc{
+					resource.TestCheckResourceAttr(resourceName, "ocsp_disable", "false"),
+					resource.TestCheckResourceAttr(resourceName, "ocsp_expiry", "23h"),
+					resource.TestCheckResourceAttr(resourceName, "auto_rebuild", "true"),
+					resource.TestCheckResourceAttr(resourceName, "auto_rebuild_grace_period", "24h"),
+					resource.TestCheckResourceAttr(resourceName, "enable_delta", "true"),
+					resource.TestCheckResourceAttr(resourceName, "delta_rebuild_interval", "18m"),
+				}
+			}
+			checks = append(checks, extras...)
+		} else {
+			checks = baseChecks
+		}
+		return resource.ComposeAggregateTestCheckFunc(checks...)(state)
+	}
+}
 
+func TestPkiSecretBackendCrlConfig_lte_111(t *testing.T) {
+	setupCRLConfigTest(t, func() {
+		testutil.TestAccPreCheck(t)
+		SkipIfAPIVersionLT(t, testProvider.Meta(), provider.VaultVersion112)
```

Comment:
Should this be `SkipIfAPIVersionGTE`? Looks like we want this test suite to run for Vault versions 1.11 and lower, is that correct?

Reply:
Good catch! Should be fixed in 56451ca115b82780d3cfc84a89650025317fff55